packages/react/src/utils/useTransitionStatus.ts

```diff
@@ -8,9 +8,12 @@ export type TransitionStatus = 'entering' | 'exiting' | undefined;
  * Provides a status string for CSS animations.
  * @param open - a boolean that determines if the element is open.
  * @param enabled - a boolean that determines if the logic is enabled.
+ * @param delayStartingStatus - a boolean  that set the `entering` status one
+ *     tick later. For edge cases e.g. collapsible needs an extra frame in order
+ *     to measure the panel contents.
  * @ignore - internal hook.
  */
-export function useTransitionStatus(open: boolean, enabled = true) {
+export function useTransitionStatus(open: boolean, enabled = true, delayStartingStatus = false) {
```

Comment:
@atomiks or `delayEnteringStatus`? If we're not renaming it internally https://github.com/mui/base-ui/pull/877

Reply:
Yeah I haven't renamed it internally, so we should match `entering`

Reply:
Updated ~